src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1475 @@
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] => 0;
+    public int M() => P;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+    public int M(int x) => this[x];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public EventHandler P => e;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsSubscribed()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|e|];
+    public void M()
+    {
+        e += MyHandler;
+    }
+
+    static void MyHandler(object sender, EventArgs e)
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsRaised()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System;
+
+class MyClass
+{
+    private event EventHandler [|_eventHandler|];
+
+    public void RaiseEvent(EventArgs e)
+    {
+        _eventHandler(this, e);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldInNameOf()
+        {
+            // https://github.com/dotnet/roslyn/issues/29519 tracks flagging and fixing this appropriately.
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    private string _goo2 = nameof(_goo);
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldInDocComment()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+/// <summary>
+/// <see cref=""C._goo""/>
+/// </summary>
+class C
+{
+    private static int [|_goo|];
+}",
+@"
+/// <summary>
+/// <see cref=""C._goo""/>
+/// </summary>
+class C
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOnlyWritten()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public void M()
+    {
+        _goo = 0;
+    }
+}",
+@"class MyClass
+{
+    public void M()
+    {
+        _goo = 0;
```

Comment:
this has broken the user's code.  we should not be doing this.

At best, if we want to offer this, it needs to be a separate feature for "remove only-written, but not read, members". And we need to properly understand the difference and warn/error if values are initialized with code that can have side effects.  

For example, say this was `_goo = ImportantCodeThatMustRun();`, while we could consider removing _goo, we would have to do so in a way that would both not cause the compile to fail, and would not cause that method to *not* be executed.